### PR TITLE
Fixes of Type Hinting in Bolt\Storage\Query namespace

### PR DESCRIPTION
**Pull request by gander**

### CODE REVIEW

File: src/Storage/Query/ContentQueryParser.php [hunk 1]
```diff
@@ -102,7 +103,7 @@ public function setQuery($query)
      *
      * @param array $params
      */
-    public function setParameters($params)
+    public function setParameters(array $params)
```
Comment (GawainLynch) [inline src/Storage/Query/ContentQueryParser.php]: @rossriley Do we want to be changing type hints on a stable branch?

File: src/Storage/Query/Handler/SelectQueryHandler.php [hunk 2]
```diff
@@ -15,7 +16,7 @@ class SelectQueryHandler
     /**
      * @param ContentQueryParser $contentQuery
      *
-     * @return QueryResultset
+     * @return QueryResultset|Content|null
```
Comment (GawainLynch) [inline src/Storage/Query/Handler/SelectQueryHandler.php]: That is not correct.

File: src/Storage/Query/ContentQueryParser.php (hunk 1 above)
Comment (gander) [inline src/Storage/Query/ContentQueryParser.php]: `$params` is array type, require the correct data type

File: src/Storage/Query/Handler/SelectQueryHandler.php (hunk 2 above)
Comment (gander) [inline src/Storage/Query/Handler/SelectQueryHandler.php]: it's return `QueryResultset` or `Content`  or `null`. Find better, but don't set one type. Maybe mixed?

Comment (GawainLynch) [inline src/Storage/Query/Handler/SelectQueryHandler.php]: Do you have test cases or is this from the IDE?

Comment (gander) [inline src/Storage/Query/Handler/SelectQueryHandler.php]: 1) dump output for certain queries
2) its used for queries so should return set or content or null
3) look at line 51 & 53

**Comment by GawainLynch** [comment]
@gander This was discussed at tonight's developer meeting, and we're happy to run with it. Only caveat being we'll merge it to 3.4 (probably in the morning), so as always :+1:

**Comment by GawainLynch** [comment]
Merged into `release/3.4` via d7388ce795b3525edde16e3def4093ccd6cd1fdf

@gander as you submitted this from a release branch, you will need to hard reset that branch. If you're stuck with that, call out on IRC and I'll happily help :smile:

**Pull request closed without merging**